### PR TITLE
Remove duplicate code in udisksata.c and udiskdaemonutil.c

### PR DESCRIPTION
**Pull request by tasleson**
Couple of smaller fixes for duplicate code in `udiskstat.c` and `udiskdaemonutil.c`

Changes are based on `master-libblockdev` branch.  Changes tested with local `make ci`.

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

File: src/udisksata.c [hunk 1]
```diff
@@ -57,6 +57,34 @@
  */
 
 /**
+ * Set the CDB bytes 1 & 2 for correct SATA protocol
+ * @param cdb  CDB to be updated (bytes 1 & 2)
+ * @param protocol Specified protocol
```
Comment (vpodzime) [inline src/udisksata.c]: Please use the ``gtk-doc`` format of documentation comments.

File: src/udisksdaemonutil.c [hunk 2]
```diff
@@ -879,6 +879,60 @@ udisks_daemon_util_check_authorization_sync_with_error (UDisksDaemon           *
 
 /* ---------------------------------------------------------------------------------------------------- */
 
+static gboolean
+dbus_freedesktop_guint32_get ( GDBusMethodInvocation   *invocation,
```
Comment (vpodzime) [inline src/udisksdaemonutil.c]: Why the space after ``(``?

File: src/udisksdaemonutil.c [hunk 3]
```diff
@@ -904,48 +958,18 @@ udisks_daemon_util_get_caller_uid_sync (UDisksDaemon            *daemon,
...
-                                       "GetConnectionUnixUser", /* method */
-                                       g_variant_new ("(s)", caller),
-                                       G_VARIANT_TYPE ("(u)"),
-                                       G_DBUS_CALL_FLAGS_NONE,
-                                       -1, /* timeout_msec */
-                                       cancellable,
-                                       &local_error);
-  if (value == NULL)
+  if (!dbus_freedesktop_guint32_get(invocation, cancellable, "GetConnectionUnixUser",
+                   &uid, error))
```
Comment (vpodzime) [inline src/udisksdaemonutil.c]: Please add a space after the function name and indent the follow-up line under the first parameter.

File: src/udisksdaemonutil.c [hunk 4]
```diff
@@ -1006,55 +1030,14 @@ udisks_daemon_util_get_caller_pid_sync (UDisksDaemon            *daemon,
...
-  g_variant_get (value, "(u)", &pid);
-  if (out_pid != NULL)
-    *out_pid = pid;
-
-  ret = TRUE;
-
- out:
-  return ret;
+  return dbus_freedesktop_guint32_get(invocation, cancellable, "GetConnectionUnixProcessID",
+                     (guint32*)(out_pid), error);
```
Comment (vpodzime) [inline src/udisksdaemonutil.c]: Same here.

**Comment by vpodzime** [comment]
I checked the test failures and none of them is related to the changes being done here.

File: src/udisksdaemonutil.c (hunk 2 above)
Comment (tasleson) [inline src/udisksdaemonutil.c]: Corrected

**Comment by tasleson** [comment]
@vpodzime All suggested changes completed.

File: src/udisksata.c [hunk 5]
```diff
@@ -57,6 +57,36 @@
...
 
 /**
+ * sata_protocol:
+ * @cdb: The CDB byte array to be updated (bytes 1 & 2)
+ * @protocol: The specified protocol
+ *
+ * Set the CDB bytes 1 & 2 for correct SATA protocol.
+ */
+static void
+sata_protocol(guint8 cdb[], UDisksAtaCommandProtocol protocol)
```
Comment (vpodzime) [inline src/udisksata.c]: Missing space after the name function.

Comment (tasleson) [inline src/udisksata.c]: @vpodzime Space added, thanks!

**Pull request merged**